File: python/pad_array.py
#REMEMBER TO PSEUDOCODE
def pad(array, min_size, value = None):

    padCount = 0
    answer = ""
    fillLen = min_size - len(array)
    if fillLen > 1:
        while padCount < fillLen:
            array.append(value)
            padCount += 1
    
    answer = array
    return answer

print(pad([1,2,3],7,"p"))
print(pad(["w","w","w"],10,"e"))
print(pad(["fox","box"],2,"p"))
print(pad(["fox", "box","fox", "box"],8,"sox!"))
print(pad([1,"two",3,"four"],12,"number"))

# look into rjust, ljust as python padding methods

File: js/padArray.js
// REMEMBER TO PSEUDOCODE
exports.pad = function(array, minSize, value=null) {
    
    let padCount = 0;
    let fillLen = minSize - array.length;
    let answer = '';
    if (fillLen < 1) {
        answer = array;
    } else {
        while (padCount < fillLen) {
            array.push(value);
            padCount++;
        };
        answer = array;
    };
    return answer;
}



// not sure why .padEnd will not work, tried on several sites

/*

if (array.length < minSize) {
      let padStr = array.join("").toString();
      padStr.padEnd(minSize, value);
      return padStr;
    } else {
    return array;
    }

*/